pkg/broker/handler/pool/pool.go

```diff
@@ -27,25 +30,86 @@ type SyncPool interface {
 	SyncOnce(ctx context.Context) error
 }
 
+type healthChecker struct {
+	mux              sync.RWMutex
+	lastReportTime   time.Time
+	maxStaleDuration time.Duration
+}
+
+func (c *healthChecker) reportHealth() {
+	c.mux.Lock()
+	defer c.mux.Unlock()
+	c.lastReportTime = time.Now()
+}
+
+func (c *healthChecker) lastTime() time.Time {
+	c.mux.RLock()
+	defer c.mux.RUnlock()
+	return c.lastReportTime
+}
+
+func (c *healthChecker) start(ctx context.Context) {
+	c.reportHealth()
+	// TODO: allow changing port? It's pure internal though.
+	srv := &http.Server{
+		Addr:    ":8080",
+		Handler: c,
+	}
+
+	go func() {
+		logging.FromContext(ctx).Info("Starting the sync pool health checker...")
```

Comment:
Maybe better as Debug?

Reply:
I think this is only called once so an Info is OK.

Reply:
Yes, it will be once.